### ISSUE TITLE
Should skip over package-private accessor in case class

### ISSUE THREAD

**Issue by adpi2** [scalacenter/scala-debug-adapter#332]
For instance, in the standard library,
```scala
final case class :: [+A](override val head: A, private[scala] var next: List[A @uncheckedVariance])
```

The method `$colon$colon.next$access$1()` is the getter of field `next` and it should be skipped.

## How to fix

In `ScalaStepFilter` we can skip methods matching `$access$\d+`.

Can the field also be lazy? What should we do?

**Comment by iusildra** [comment]
I think it has already been done somewhere. With the following code 
```scala
package ThereIsNoSpoon

case class Test(private[ThereIsNoSpoon] var test: Int)

object Main {
  def main(args: Array[String]): Unit = {
    val test = Test(1)
    test.test
    println("ok")
  }
}
```
I can't "Step into" the accessor with Scala 2.13 (it has been removed in Scala 3)

**Comment by adpi2** [comment]
> I think it has already been done somewhere.

It surprises me a bit. I would expect `next$access$1` to be interpreted as a local method and to never be skipped. I'll take a closer look later.

**Comment by iusildra** [comment]
Nah, I saw it in the wrong place, when I didn't know ScalaStepFilter was renamed to ScalaUnpickler. Forgot to update it
